integration/jnifuse/native/src/main/native/libjnifuse/jnifuse_jvm.cc

```diff
@@ -0,0 +1,100 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+#include <jni.h>
+#include <pthread.h>
+
+#include "check.h"
+#include "debug.h"
+
+namespace jnifuse {
+
+static JavaVM* g_jvm = nullptr;
+
+// Key for per-thread JNIEnv* data. Non-NULL in threads attached to `g_jvm` by
+// AttachCurrentThreadIfNeeded(), NULL in unattached threads and threads that
+// were attached by the JVM because of a Java->native call.
+static pthread_key_t g_jni_ptr;
+
+JavaVM* GetJVM() {
+  JNIFUSE_CHECK(g_jvm, "Failed to run JNI_OnLoad?");
+  return g_jvm;
+}
+
+// Return a |JNIEnv*| usable on this thread or NULL if this thread is detached.
+JNIEnv* GetEnv() {
```

Comment:
GetEnv called frequently? Or just once?

Reply:
Any possible performance issue?

Reply:
If it is expensive to get from g_jvm, is it possible to fetch it from pthread_getspecific

Reply:
The implementation in this PR is safer, it gives us more information about the status of the current thread.
But it's likely to have performance issues since it's be called in each small fuse operations.
I will change to fetch it from `pthread_getspecific`